### ISSUE TITLE
Navigation.events().registerAppLaunchedListener never been triggered

### ISSUE THREAD

**Issue by JoseFMP** [wix/react-native-navigation#4657]
### Issue Description

The event 
```ts
Navigation.events().registerAppLaunchedListener
```
is never trieggered.

### Steps to Reproduce / Code Snippets / Screenshots
* Create an empty react-native application
* In the entry of your application, subscribe to the `registerAppLaunchedListener` event.
* Run the app.
`index.js`:
```ts
import { componentGenerator } from './screens/LandingScreen'
import { Navigation } from 'react-native-navigation';

console.log('Trying to subscribe to AppLaunched event'); // this line is hit
Navigation.registerComponent('LandingScreen', componentGenerator);
Navigation.events().registerAppLaunchedListener(() => {
   console.log('Hello world'); // this line is never hit
   Navigation.setRoot({
            root: {
                component: {
                    name: 'LandingScreen', 
                    id: 'LandingScreen',
                }
            }
   });
});
console.log('Finished subscribing to event'); // this line is hit
```
The callback for the event `registerAppLaunchedListener` is never called.
Expected behavior: The callback is executed

---
### Environment
* React Native Navigation version: 0.58.1
* React Native version: 2.8.0
* Platform(s) (iOS, Android, or both?): Both
* Device info (Simulator/Device? OS version? Debug/Release?): LG V20

**Comment by jinshin1013** [comment]
Is the other console.log being called? Can you share the full index.js or your entry file?

**Comment by JoseFMP** [comment]
@jinshin1013  I updated it in the original message. So whatever is not related to the event is working fine. But the event is not triggered.

**Comment by jinshin1013** [comment]
Do you get any errors? or just blank screen?

**Comment by JoseFMP** [comment]
@jinshin1013  I get no error, just blank screen and the event no triggered (i.e. debugger does not stop in break points in the callback, the log messages do not appear, I also tried to throw errors in the block to see if they would appear).

**Comment by jinshin1013** [comment]
It may sound stupid but you did follow all the instruction to install react-native-navigation, right? If you could prep up a reproducible project on your Github, I could have a look at it.

**Comment by jinshin1013** [comment]
It may be due to react-native-navigation not supporting RN ^0.58. Try downgrading to 0.57.8 and see if it builds.

**Comment by bobmulder** [comment]
I had the same and saw your issue. The mistake I made was that the listener was registered too early. Try to register the listener (with a `console.log`) as soon as possible. Hope it works for you!

**Comment by dmchoull** [comment]
@bobmulder Can you rephrase or elaborate on what your solution was? I'm not clear on what it means to register the listener too early or how you changed that. Are you doing something different than the [example in the documentation](https://wix.github.io/react-native-navigation/#/docs/Installing?id=you-can-use-react-native-navigation-o) (which is not working for me, even after downgrading RN to 0.57.8)

**Comment by marf** [comment]
Same problem here, in iOS it works, while in Android the event is never triggered.

@bobmulder @dmchoull Have you found a solution to the problem?

**Comment by dmchoull** [comment]
@marf No, unfortunately not. I've just switched to using react-navigation for now

**Comment by TheBurgerShot** [comment]
I was experiencing the same issue. On iOS the event was fired, on Android it wasn't.

In my case, setting the registerAppLaunchedListener before the promised did the trick.

So changing this:


```
Promise.all([
        Icon.getImageSource(Platform.OS === 'android' ? "md-map" : "ios-map", 30),
        Icon.getImageSource(Platform.OS === 'android' ? "md-list" : "ios-list", 30),
        Icon.getImageSource(Platform.OS === 'android' ? "md-more" : "ios-more", 30)
]).then(sources => {
      console.log('Loaded sources'); // <-- This one does fire on android
      Navigation.events().registerAppLaunchedListener(() => {
          console.log('App launched'); // <-- This one doesn't fire on android
           ...
      });
})
```

to this:

```
Navigation.events().registerAppLaunchedListener(() => {
    Promise.all([
        Icon.getImageSource(Platform.OS === 'android' ? "md-map" : "ios-map", 30),
        Icon.getImageSource(Platform.OS === 'android' ? "md-list" : "ios-list", 30),
        Icon.getImageSource(Platform.OS === 'android' ? "md-more" : "ios-more", 30)
    ]).then(sources => {
        console.log('Loaded sources'); // <-- This one does fire on android
    })
})
```

Your case might be different. But this did the trick for me. Hope it helps.

**Comment by ShaneMatthias** [comment]
"Register the listener with registerAppLaunchedListener as soon as possible - it should be one of the first lines in your index.js file. If you're observing a "white screen" or a hanging splash screen after relaunching your app, it probably means Navigation.setRoot isn't called once the app has launched. Perhaps the listener was registered too late." - From the docs
